alica_tests/src/test/test_blackboard.cpp

```diff
@@ -34,68 +36,161 @@ class NonConstructable
+}
+
+TEST_F(SingleAgentTestFixture, testValueMappingPlans)
+{
+    // Checks if value mapping for conditions succeeds
+
+    // Transition to the plan corresponding to this test case
+    ASSERT_TRUE(_tc->setTransitionCond("TestMasterPlan", "ChooseTestState", "BlackboardTestState")) << _tc->getLastFailure();
+    STEP_UNTIL(_tc, _tc->getActivePlan("TestBlackboardPlan"));
+    auto plan = _tc->getActivePlan("TestBlackboardPlan");
+    ASSERT_NE(plan, nullptr) << _tc->getLastFailure();
+
+    ASSERT_TRUE(_tc->setTransitionCond("TestBlackboardPlan", "SelectBlackboardTestState", "ValueMappingPlansTestState")) << _tc->getLastFailure();
+    plan = _tc->getActivePlan("TestBlackboardPlan");
+    ASSERT_NE(plan, nullptr) << _tc->getLastFailure();
+
+    // Step until the plan succeeds
+    STEP_UNTIL_ASSERT_TRUE(_tc, _tc->isSuccess(plan));
+}
 
-TEST_F(TestBlackboard, testJsonTwoBehaviorKeyMapping)
+TEST_F(SingleAgentTestFixture, testValueMappingKeyNotFound)
+{
+    BlackboardBlueprint targetBlueprint;
+    targetBlueprint.addKey("targetKey", BBType::BOOL);
+
+    Blackboard targetBB(&targetBlueprint);
+    EXPECT_THROW({ targetBB.impl().mapValue("wrongTargetKey", "true"); }, BlackboardException);
+}
+
+TEST_F(SingleAgentTestFixture, testValueMappingUnknownType)
+{
+    BlackboardBlueprint targetBlueprint;
+    targetBlueprint.addKey("targetKey", BBType::ANY);
+
+    Blackboard targetBB(&targetBlueprint);
+    EXPECT_THROW({ targetBB.impl().mapValue("targetKey", "test"); }, BlackboardException);
+}
+
+TEST_F(SingleAgentTestFixture, testValueMappingCantParseValue)
+{
+    BlackboardBlueprint targetBlueprint;
+    targetBlueprint.addKey("targetKey", BBType::BOOL);
+
+    Blackboard targetBB(&targetBlueprint);
+    EXPECT_THROW({ targetBB.impl().mapValue("targetKey", "test"); }, BlackboardException);
+}
+
+TEST_F(SingleAgentTestFixture, testJsonTwoBehaviorKeyMapping)
 {
     // Two parent values mapped to same behavior input value to differentiate call contexts
-    ae->start();
-    ae->getAlicaClock().sleep(alica::AlicaTime::milliseconds(200));
 
-    std::shared_ptr<alicaTests::TestWorldModel> wm =
-            LockedBlackboardRW(ae->editGlobalBlackboard()).get<std::shared_ptr<alicaTests::TestWorldModel>>("worldmodel");
-    EXPECT_EQ(wm->passedParameters["behaviorInputKey"], 5);       // Value set in first behavior call
-    EXPECT_EQ(wm->passedParameters["behaviorSecondInputKey"], 7); // Value set in second behavior call
+    // Transition to the plan corresponding to this test case
+    ASSERT_TRUE(_tc->setTransitionCond("TestMasterPlan", "ChooseTestState", "BlackboardTestState")) << _tc->getLastFailure();
+    STEP_UNTIL(_tc, _tc->getActivePlan("TestBlackboardPlan"));
+    auto plan = _tc->getActivePlan("TestBlackboardPlan");
+    ASSERT_NE(plan, nullptr) << _tc->getLastFailure();
+
+    ASSERT_TRUE(_tc->setTransitionCond("TestBlackboardPlan", "SelectBlackboardTestState", "JsonTwoBehaviorKeyMappingTestState")) << _tc->getLastFailure();
```

Comment:
I don't understand what's going on with these testjson* tests or the empty states in TestBlackboard Plan